File: jeehyun/개별문제/SW Expert Academy/1225_암호생성기.cpp
#include <iostream>
#include <queue>

using namespace std;

int main(int argc, char** argv)
{
    int k, T = 10;
    queue<int> q;

    for (int test_case = 1; test_case <= T; ++test_case) {
        cin >> test_case;

        for (int i = 0; i < 8; i++) {
            cin >> k;
            q.push(k);
        }

        int cycle = 1;
        while (1) {
            int now = q.front() - cycle;
            q.pop();
            
            // 0보다 작아지거나 0일 경우 0 저장하고 종료
            if (now <= 0) {
                q.push(0);
                break;
            }
            q.push(now);
            cycle++;

            // 사이클이 끝나면 다시 1부터 시작
            if (cycle > 5)
                cycle = 1;
        }

        cout << "#" << test_case << " ";

        // 암호 출력
        while (!q.empty()) {
            cout << q.front() << " ";
            q.pop();
        }
        cout << "\n";
    }
    return 0;
}

File: jeehyun/개별문제/SW Expert Academy/1225_암호생성기.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;

class Solution {
	private static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
	private static StringTokenizer st;
	private static Queue<Integer> q = new LinkedList<>();

	public static void main(String args[]) throws Exception {
		int T = 10;
		
        for (int test_case = 1; test_case <= T; test_case++) {
        	test_case = Integer.parseInt(br.readLine());
        	st = new StringTokenizer(br.readLine(), " ");
        	
        	while (st.hasMoreTokens()) {
        		q.add(Integer.parseInt(st.nextToken()));
        	}
        	
        	int cycle = 1;
        	while (true) {
        		int now = q.poll() - cycle;
        		
        		if (now <= 0) break;
        		q.add(now);
        		
        		cycle++;
        		if (cycle > 5) cycle = 1;
        	}
        	q.add(0);
        	
        	System.out.print("#" + test_case + " ");
        	while (!q.isEmpty()) {
        		System.out.printf("%d ", q.poll());
        	}
        	System.out.println();
		}
	}
}